### PR TITLE
byteorder: add helpers to for host <-> little endian

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR adds helpers for converting between host and little endian. These functions are useful for frames using little endian (e.g LoRaWAN).

I also added the corresponding unittests.

### Testing procedure
`make -C tests/unittests tests-core test`

### Issues/PRs references
Came from #14058

### CODE REVIEW

**Comment by bergzand** [comment]
Duplicate! #10986 :laughing: 

But feel free to take over with this PR

File: core/include/byteorder.h [hunk 1]
```diff
@@ -203,6 +203,48 @@ static inline uint32_t byteorder_ntohl(network_uint32_t v);
  */
 static inline uint64_t byteorder_ntohll(network_uint64_t v);
 
+/**
+ * @brief          Convert from host byte order to little endian , 16 bit.
+ * @param[in]      v   The integer in host byte order.
+ * @returns        `v` converted to little endian.
+ */
+static inline network_uint16_t byteorder_htols(uint16_t v);
```
Comment (miri64) [inline core/include/byteorder.h:211]: Here and basically everywhere: Type confusion ;-).

```suggestion
static inline le_uint16_t byteorder_htols(uint16_t v);
```

**Comment by jia200x** [comment]
> Duplicate! #10986 laughing

:O!!
My fault! Sorry, didn't check that one.

I think #10986 is in better shape. I will take over then, thanks!

**Comment by aabadie** [comment]
#15519 was merged and this one is a duplicate. Closing.

**Pull request closed without merging**